src/app/pages/general/general.js

```diff
@@ -1,45 +1,50 @@
-import componentType, {canonicalLinkMixin} from '~/helpers/controller/init-mixin';
-import css from './general.css';
-import $ from '~/helpers/$';
+import {pageWrapper} from '~/controllers/jsx-wrapper';
+import React from 'react';
+import {LoaderPage, RawHTML} from '~/components/jsx-helpers/jsx-helpers.jsx';
 import {urlFromSlug} from '~/models/cmsFetch';
+import './general.css';
 
-const spec = {
-    css,
-    view: {
-        classes: ['general', 'page'],
-        tag: 'main'
-    }
-};
-const BaseClass = componentType(spec, canonicalLinkMixin);
+function GeneralPage({html}) {
+    const parser = new DOMParser();
+    const newDoc = parser.parseFromString(html, 'text/html');
+    const strips = parser
+        .parseFromString(
+            '<img class="strips" src="/images/components/strips.svg" height="10" alt="">',
+            'text/html'
+        )
+        .querySelector('img');
+    const innerHTML = Array.from(newDoc.body.children).reduce((arr, el) => {
+        if (el.classList.contains('block-heading')) {
+            el.appendChild(strips);
+        }
+        arr.push(el.outerHTML);
+        return arr;
+    }, []).join('\n');
+
+    return (
+        <RawHTML className="general page" html={innerHTML} embed />
+    );
+}
```

Comment:
hm, I haven't used `DOMParser` before, this is pretty neat.  Do you fully trust the html though? if not this looks like it could be a big security risk since it allows rendering arbitrary html

Reply:
This is for HTML pulled from the server, so yes. I would have just used RawHTML, but I had to identify and embellish the .block-heading piece.